<?xml version="1.0" encoding="utf-8"?>

<launch>

  <arg name="vessel_ns"       default="r2sonic"/>
  <arg name="vessel_config"   default="r2sonic"/>

  <node pkg="r2sonic" exec="r2sonic_node" name="mbes" namespace="$(var vessel_ns)/perception/sensors">
    <param from="$(find-pkg-share r2sonic)/config/r2sonic.yaml"/>
    <param from="$(find-pkg-share $(var vessel_config))/config/r2sonic.yaml"/>
    <param name="tx_frame_id" value="$(var vessel_ns)/acoustic_center"/>
    <param name="rx_frame_id" value="$(var vessel_ns)/acoustic_center"/>
  </node>

  <node pkg="projectron" exec="basic_projector" name="basic_projector" namespace="$(var vessel_ns)/perception/sensors/mbes">
  </node>

  <node pkg="sbg_driver" exec="sbg_device" name="sbg_mru" namespace="$(var vessel_ns)/nav/sensors/mru">
    <param from="$(find-pkg-share r2sonic)/config/sbg_mru.yaml"/>
    <param from="$(find-pkg-share $(var vessel_config))/config/sbg_mru.yaml"/>
  </node>

  <node pkg="mru_transform" exec="mru_transform_node" name="mru_transform" output="screen" namespace="$(var vessel_ns)/nav">
    <param from="$(find-pkg-share r2sonic)/config/mru_transform.yaml"/>
    <param from="$(find-pkg-share $(var vessel_config))/config/mru_transform.yaml"/>
    <param name="base_frame" value="$(var vessel_ns)/base_link"/>
  </node>

  <include file="$(find-pkg-share r2sonic)/launch/publish_urdf.launch.py">
    <arg name="vessel_ns"     value="$(var vessel_ns)"/>
  </include>






</launch>
